src/wallet/rpcwallet.cpp

```diff
@@ -2008,6 +2016,9 @@ Value encryptwallet(const Array& params, bool fHelp)
 
     if (fHelp)
         return true;
+    if (!fEnableWalletEncryption) {
+        return false;
```

Comment:
The observable behaviour is that the call returns `false` (instead of `true`). Agreed that throwing an error is more explicit.